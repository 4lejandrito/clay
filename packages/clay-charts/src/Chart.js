import ChartBase from './ChartBase';
import DataComponent from './DataComponent';
import Soy from 'metal-soy';
import templates from './Chart.soy.js';

/**
 * Metal Chart component.
 * @extends ChartBase
 * @extends Component
 */
class Chart extends DataComponent {}

Object.setPrototypeOf(ChartBase, DataComponent.prototype);
Object.assign(Chart.prototype, ChartBase);
Chart.STATE = Object.assign({}, ChartBase.STATE);

Soy.register(Chart, templates);

export {Chart};
export default Chart;
