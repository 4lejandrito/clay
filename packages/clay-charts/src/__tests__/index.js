import React from 'react';
import TestRenderer from 'react-test-renderer';
import Chart, {BarChart} from '../index';

const COLUMNS = [['data1', 100, 20, 30], ['data2', 20, 70, 100]];

describe('Chart', function() {
	it('should render named chart without js exception', () => {
		const testRenderer = TestRenderer.create(
			<BarChart
				className="bar-chart"
				data={{
					columns: COLUMNS,
				}}
			/>
		);

		expect(testRenderer.toJSON()).toMatchSnapshot();
	});

	it('should render with type without js exception', async () => {
		const testRenderer = TestRenderer.create(
			<Chart
				className="bar-chart"
				data={{
					columns: [['data1', 30], ['data2', 70]],
					type: 'donut',
				}}
			/>
		);

		expect(testRenderer.toJSON()).toMatchSnapshot();
	});
});
