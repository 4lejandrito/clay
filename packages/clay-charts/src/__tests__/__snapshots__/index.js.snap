// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Chart should render named chart without js exception 1`] = `
<div
  className="bar-chart"
  style={undefined}
/>
`;

exports[`Chart should render with type without js exception 1`] = `
<div
  className="bar-chart"
  style={undefined}
/>
`;
